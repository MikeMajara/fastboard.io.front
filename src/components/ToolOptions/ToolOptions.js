import React, { useRef } from 'react';
// import { Circle as CircleIcon } from '@styled-icons/fa-solid/Circle';

// import ToolButton from '../ToolButton';

import SWrapper from './styled';

const ToolOptions = ({ show, children }) => {
  return (
    <SWrapper show={show}>
      {children}
    </SWrapper>
  );
};

export default ToolOptions;
