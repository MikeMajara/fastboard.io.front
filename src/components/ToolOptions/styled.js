import styled, { css } from '@xstyled/styled-components';
import {
  breakpoints, getBorderWidth, getSize, getSpace, variant,
} from '@xstyled/system';

export default styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  transition: fast;

  ${breakpoints({
    xs : css`top: calc(2 * ${getSpace('xxs')} + ${getSize('toolButtonSm')} + 2 * ${getBorderWidth('default')} + ${getSpace('xl')});`,
    sm : css`top: calc(2 * ${getSpace('xxs')} + ${getSize('toolButtonMd')} + 2 * ${getBorderWidth('default')} + ${getSpace('xl')});`,
    xl : css`top: calc(2 * ${getSpace('sm')} + ${getSize('toolButtonLg')} + 2 * ${getBorderWidth('default')});`,
  })}

  ${variant({
    default  : false,
    prop     : 'show',
    variants : {
      true : css`
        opacity : 1;
        visibility: visible;
      `,
      false : css`
        opacity : 0;
        transform: translate(0, -10%);
        visibility: hidden;
        pointer-events:none;
      `,
    },
  })}

  & > div + div {
    margin-top: 6px;
  }
`;
