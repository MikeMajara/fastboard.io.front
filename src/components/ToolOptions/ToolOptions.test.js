import React from 'react';
import { render } from '@testing-library/react';

import ToolOptions from './ToolOptions';

describe('Component : ToolOptions', () => {
  test('Is visible and shows children', () => {
    const childrenTestId = 'tooloptions-children';
    const children = <span data-testid={childrenTestId} />;
    const { getByTestId } = render(
      <ToolOptions show>
        {children}
      </ToolOptions>,
    );

    expect(getByTestId(childrenTestId)).toBeInTheDocument();
  });

  test('Snapshot', () => {
    const { asFragment } = render(
      <ToolOptions show>
        <span />
      </ToolOptions>,
    );

    const componentRender = asFragment();
    expect(componentRender).toMatchSnapshot();
  });
});
