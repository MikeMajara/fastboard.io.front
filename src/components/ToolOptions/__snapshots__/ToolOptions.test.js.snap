// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component : ToolOptions Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-pscky fwiqed"
  >
    <span />
  </div>
</DocumentFragment>
`;
